pkg/cmd/pulumi/new_ai.go

```diff
@@ -0,0 +1,263 @@
+		isHTTPBackend
+}
+
+// Iteratively prompt the user for input, sending their input as a prompt tp Pulumi AI
+// Stream the response back to the console, and repeat until the user is done.
+func runAINew(
+	ctx context.Context,
+	args newArgs,
+	opts display.Options,
+	backend httpstate.Backend,
+) (conversationURL string, err error) {
+	languageOptions := []string{
+		"TypeScript",
+		"JavaScript",
+		"Python",
+		"Go",
+		"C#",
+		"Java",
+		"YAML",
+	}
+	var rawLanguageSelect string
+	if args.aiLanguage == "" {
+		if err = survey.AskOne(&survey.Select{
+			Message: "Please select a language for your project:",
+			Options: languageOptions,
+		}, &rawLanguageSelect, surveyIcons(opts.Color)); err != nil {
+			return "", err
+		}
+		err = args.aiLanguage.Set(rawLanguageSelect)
+		if err != nil {
+			return "", err
+		}
+	}
+	var continuePrompt string
+	var connectionID string
+	var conversationID string
+	for continuePrompt, conversationURL, conversationID, connectionID, err = runAINewPromptStep(
+		ctx,
+		backend,
+		opts,
+		args,
+		"",
+		"",
+		"",
+		"",
+	); continuePrompt == refineSelection; continuePrompt,
+		conversationURL,
+		conversationID,
+		connectionID,
+		err = runAINewPromptStep(
+		ctx,
+		backend,
+		opts,
+		args,
+		continuePrompt,
+		connectionID,
+		conversationURL,
+		conversationID,
+	) {
+		if err != nil {
+			return "", err
+		}
+	}
+	return conversationURL, err
+}
+
+func sendPromptToPulumiAI(
+	ctx context.Context,
+	backend httpstate.Backend,
+	promptMessage string,
+	conversationID string,
+	connectionID string,
+	language httpstate.PulumiAILanguage,
+) (string, string, string, error) {
+	pulumiAIURL := env.AIServiceEndpoint.Value()
+	if pulumiAIURL == "" {
+		pulumiAIURL = "https://www.pulumi.com/ai"
+	}
+	parsedURL, err := url.Parse(pulumiAIURL)
+	if err != nil {
+		return "", "", "", err
+	}
+	requestBody := httpstate.AIPromptRequestBody{
+		Language:       language,
+		Instructions:   promptMessage,
+		ResponseMode:   "code",
+		ConversationID: conversationID,
+		ConnectionID:   connectionID,
+	}
+	fmt.Println("Sending prompt to Pulumi AI...")
+	res, err := backend.PromptAI(ctx, requestBody)
+	if err != nil {
+		return "", "", "", err
+	}
+	defer res.Body.Close()
+	reader := bufio.NewReader(res.Body)
+	fmt.Println("Pulumi AI response:")
+	for {
+		chunk, err := reader.ReadByte()
+		if err != nil && err.Error() != "EOF" {
+			return "", "", "", err
+		}
+		fmt.Print(string(chunk))
+		if err != nil && err.Error() == "EOF" {
+			// Add a newline to ensure we don't overwrite the last line of the Pulumi AI response
+			fmt.Println()
+			break
+		}
+	}
+	conversationID = res.Header.Get("x-conversation-id")
+	connectionID = res.Header.Get("x-connection-id")
+	projectURL := parsedURL.JoinPath("api", "project", url.PathEscape(fmt.Sprintf("%s.zip", conversationID))).String()
+	conversationURL := parsedURL.JoinPath("conversations", conversationID).String()
+	fmt.Println("View this conversation at:", conversationURL)
```

Comment:
Should there be a space after the colon?

```suggestion
	fmt.Printf("View this conversation at: %s\n", conversationURL)
```